src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -565,16 +566,54 @@ def rshift(self, y: int) -> "ReplicatedSharedTensor":
 
         Raises:
             ValueError: If y is not an integer.
+            ValueError : If invalid shift value is provided.
         """
         if not isinstance(y, int):
             raise ValueError("Right Shift works only with integers!")
 
+        ring_bits = get_nr_bits(self.ring_size)
+        if y > ring_bits - 1:
+            raise ValueError(
+                f"Invalid value for right shift : {y},must be in range:[0,{ring_bits-1}]"
```

Comment:
NITPICK: One empty space between `,` and `must` and remove space before `:` --> `shift:`